openmdao/core/problem.py

```diff
@@ -2369,8 +2371,11 @@ def set_later(var_name):
                                 val = case.inputs[abs_name]
                             else:
                                 val = case.inputs[case_abs_names[0]]
-
-                        varmeta = abs2meta_in[abs_name]
+                        try:
```

Comment:
I'd avoid using try/except for logic:
`varmeta = abs2meta_in[abs_name] if abs_name in abs2meta_in else abs2meta_disc_in[abs_name]`